### ISSUE TITLE
Urgent/Critical: broken sub-modules

### ISSUE THREAD

**Issue by haysclark** [jenkins-x/terraform-google-jx#209]
I can no longer `create`, `edit` or `destroy` a cluster using this module because sub-modules have had API changes. The last time I was able to use this Terraform template was 23 days ago, so clearly a sub-module has changed and the version is not locked down.

Terraform Cloud error log:

```
Terraform v0.13.7
Configuring remote state backend...
Initializing Terraform configuration...

Error: Missing required argument

  on .terraform/modules/jx/modules/backup/main.tf line 31, in resource "google_project_iam_member" "velero_sa_storage_admin_binding":
  31: resource "google_project_iam_member" "velero_sa_storage_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/backup/main.tf line 39, in resource "google_project_iam_member" "velero_sa_storage_object_admin_binding":
  39: resource "google_project_iam_member" "velero_sa_storage_object_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/backup/main.tf line 47, in resource "google_project_iam_member" "velero_sa_storage_object_creator_binding":
  47: resource "google_project_iam_member" "velero_sa_storage_object_creator_binding" {

The argument "project" is required, but no definition was found.


Error: Unsupported argument

  on .terraform/modules/jx/modules/cluster/main.tf line 33, in resource "google_container_cluster" "jx_cluster":
  33:     username = ""

An argument named "username" is not expected here.


Error: Unsupported argument

  on .terraform/modules/jx/modules/cluster/main.tf line 34, in resource "google_container_cluster" "jx_cluster":
  34:     password = ""

An argument named "password" is not expected here.


Error: Unsupported argument

  on .terraform/modules/jx/modules/cluster/main.tf line 48, in resource "google_container_cluster" "jx_cluster":
  48:     identity_namespace = "${var.gcp_project}.svc.id.goog"

An argument named "identity_namespace" is not expected here.


Error: Missing required argument

  on .terraform/modules/jx/modules/cluster/serviceaccount.tf line 14, in resource "google_project_iam_member" "build_controller_sa_storage_object_admin_binding":
  14: resource "google_project_iam_member" "build_controller_sa_storage_object_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/cluster/serviceaccount.tf line 20, in resource "google_project_iam_member" "build_controller_sa_storage_admin_binding":
  20: resource "google_project_iam_member" "build_controller_sa_storage_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/cluster/serviceaccount.tf line 34, in resource "google_project_iam_member" "kaniko_sa_storage_admin_binding":
  34: resource "google_project_iam_member" "kaniko_sa_storage_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/cluster/serviceaccount.tf line 48, in resource "google_project_iam_member" "tekton_sa_storage_object_admin_binding":
  48: resource "google_project_iam_member" "tekton_sa_storage_object_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/cluster/serviceaccount.tf line 54, in resource "google_project_iam_member" "tekton_sa_storage_admin_binding":
  54: resource "google_project_iam_member" "tekton_sa_storage_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/cluster/serviceaccount.tf line 60, in resource "google_project_iam_member" "tekton_sa_project_viewer_binding":
  60: resource "google_project_iam_member" "tekton_sa_project_viewer_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/cluster/serviceaccount.tf line 76, in resource "google_project_iam_member" "ui_sa_storage_admin_binding":
  76: resource "google_project_iam_member" "ui_sa_storage_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/cluster/serviceaccount.tf line 84, in resource "google_project_iam_member" "ui_sa_storage_object_admin_binding":
  84: resource "google_project_iam_member" "ui_sa_storage_object_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/cluster/serviceaccount.tf line 218, in resource "google_project_iam_member" "boot_sa_storage_object_admin_binding":
 218: resource "google_project_iam_member" "boot_sa_storage_object_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/gsm/main.tf line 44, in resource "google_project_iam_member" "gsm_sa_secret_accessor_binding":
  44: resource "google_project_iam_member" "gsm_sa_secret_accessor_binding" {

The argument "project" is required, but no definition was found.


Error: Unsupported attribute

  on .terraform/modules/jx/modules/vault/main.tf line 40, in resource "google_kms_crypto_key" "vault_crypto_key":
  40:   key_ring        = google_kms_key_ring.vault_keyring[0].self_link

This object has no argument, nested block, or exported attribute named
"self_link".


Error: Missing required argument

  on .terraform/modules/jx/modules/vault/main.tf line 72, in resource "google_project_iam_member" "vault_sa_storage_object_admin_binding":
  72: resource "google_project_iam_member" "vault_sa_storage_object_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/vault/main.tf line 80, in resource "google_project_iam_member" "vault_sa_cloudkms_admin_binding":
  80: resource "google_project_iam_member" "vault_sa_cloudkms_admin_binding" {

The argument "project" is required, but no definition was found.


Error: Missing required argument

  on .terraform/modules/jx/modules/vault/main.tf line 88, in resource "google_project_iam_member" "vault_sa_cloudkms_crypto_binding":
  88: resource "google_project_iam_member" "vault_sa_cloudkms_crypto_binding" {

The argument "project" is required, but no definition was found.
```

**Comment by ankitm123** [comment]
This is the issue: https://github.com/hashicorp/terraform-provider-google/pull/10394

**Comment by ankitm123** [comment]
/close
fixed by #211

**Comment by ankitm123** [comment]
The fix is not yet merged into master ...
